backend/de.metas.manufacturing.rest-api/src/main/java/de/metas/manufacturing/job/service/ManufacturingJobLoaderAndSaver.java

```diff
@@ -387,4 +398,22 @@ public void saveHeader(@NonNull final ManufacturingJob job)
 		ppOrder.setCurrentScaleDeviceId(job.getCurrentScaleDeviceId() != null ? job.getCurrentScaleDeviceId().getAsString() : null);
 		InterfaceWrapperHelper.saveRecord(ppOrder);
 	}
+
+	@NonNull
+	private Optional<ManufacturingJobActivity> toValidateLocatorInfoActivity(final @NonNull PPOrderRoutingActivity from)
+	{
+		return Optional.of(supportingServices.getValidateSourceLocatorInfo(from.getOrderId()))
+				.filter(sourceLocatorInfo -> !sourceLocatorInfo.getSourceLocatorList().isEmpty())
+				.map(sourceLocatorInfo -> prepareJobActivity(from).sourceLocatorValidate(sourceLocatorInfo).build());
+	}
+
+	@NonNull
+	private Optional<ManufacturingJobActivity> toIssueOnlyWhatWasReceivedActivity(final @NonNull PPOrderRoutingActivity from)
+	{
+		return getBOMLines(from.getOrderId())
+				.stream()
+				.filter(bomLine -> BOMIssueMethod.Planning.getCode().equals(bomLine.getIssueMethod()))
+				.findFirst()
```

Comment:
it's hard to understand wht's the point here.
IMHO from getBOMLines(...) until `.findFirst()` can be extracted to some method which returns boolean...

Reply:
what's the conclusion here?

Reply:
the code doesn't exist anymore :)